### DISCUSSION TITLE
Bring back ICA decompositions of BOLD?

### DISCUSSION THREAD

**Discussion by oesteban** [nipreps/mriqc#1329] [Ideas]
We dropped FSL (and MELODIC) some time ago for many reasons (the principal one is FSL's noncommercial limitations). However, the problems with MELODIC were not limited to the FSL's license (see #671).

This discussion is open to gauge the interest in bringing back some ICA decomposition (perhaps based on GIFT as Chris G. proposed, or maybe with nilearn leveraging the experience of @tsalo, @emdupre and @eilidhmacnicol on this front).

**Comment by tsalo** [comment]
I'm not really familiar with using ICA for QC rather than as a denoising step, so take my thoughts with a grain of salt. In #1106 it sounds like the ICA wasn't really adding much to MRIQC, except for the possibility of useful component-derived IQMs in the future. If that is indeed the case, then I think it would be best to have a list of IQMs in mind before adding ICA back in. Something like the amount of variance explained by rejected components from an ICA-AROMA classification might be cool, but I'm not sure it's useful enough to merit adding MELODIC back in.

FWIW, I'm very interested in getting ICA tools working as post-processing BIDS Apps specifically for denoising purposes (e.g., [`fMRIPost-AROMA`](https://github.com/nipreps/fmripost-aroma), [`fMRIPost-tedana`](https://github.com/ME-ICA/fmripost-tedana)). GIFT would be a great addition- especially for complex-valued data, as in [`fMRIPost-phase`](https://github.com/nipreps/fmripost-phase).

**Reply by oesteban** [reply]
> In #1106 it sounds like the ICA wasn't really adding much to MRIQC, except for the possibility of useful component-derived IQMs in the future. If that is indeed the case, then I think it would be best to have a list of IQMs in mind before adding ICA back in. Something like the amount of variance explained by rejected components from an ICA-AROMA classification might be cool, but I'm not sure it's useful enough to merit adding MELODIC back in.

Yes, I fully agree with the assessment and the following-up idea.


> GIFT would be a great addition

Absolutely.
